lib/routes/account.js

```diff
@@ -946,10 +1013,33 @@ module.exports = function (
         }
 
         function createResponse() {
-          return {
-             email: sessionToken.email,
-             verified: sessionToken.emailVerified
-           }
+
+          var doSigninConfirmation = requestHelper.shouldEnableSigninConfirmation(sessionToken.uid.toString('hex'), config)
+
+          if (doSigninConfirmation) {
+            return db.sessionTokenWithVerificationStatus(sessionToken.id)
+              .then(function (token) {
+
+                var sessionVerified = token.tokenVerified
```

Comment:
Yea I am not sure what kind of shenanigans this would cause either lol. My initial impression is that it wouldn't cause any because it seems the only reason you would have to be calling `/recovery_email/status` is because you want to know that a session/email is verified.

I could see this tweaked a little to better maintain backwards compatibility. If the response looked like

```javascript
{
 verified: "email verified value",
 sessionVerified: "session verified value"
}
```
For future clients, they would have to make sure both are true before proceeding. Not sure if that is worthwhile or makes sense.